src/cache.rs

```diff
@@ -150,16 +150,20 @@ impl Cache {
         let client = builder
             .build()
             .context("Could not instantiate HTTP client")?;
-        let mut resp = client.get(&self.url).send()?;
+        let mut resp = client
+            .get(&self.url)
+            .send()?
+            .error_for_status()
+            .with_context(|| format!("Could not download tldr pages from {}", &self.url))?;
         let mut buf: Vec<u8> = vec![];
         let bytes_downloaded = resp.copy_to(&mut buf)?;
         debug!("{} bytes downloaded", bytes_downloaded);
         Ok(buf)
     }
 
     /// Decompress and open the archive
-    fn decompress<R: Read + Seek>(reader: R) -> ZipArchive<R> {
-        ZipArchive::new(reader).unwrap()
+    fn decompress<R: Read + Seek>(reader: R) -> Result<ZipArchive<R>> {
+        Ok(ZipArchive::new(reader)?)
     }
 
```

Comment:
Is there any benefit from having this as a method now? it would probably be easier to just inline it

Reply:
Yeah, that's true.